### DISCUSSION TITLE
[UX] Handle user trying to login in unauthorized panel

### DISCUSSION THREAD

**Discussion by underdpt** [filamentphp/filament#8222] [Help]
Hello,

Doing some tests, I tried to login with a "common" user on the admin panel, for which the user has no access. The result is that the user is shown a 403 error, but he remains logged. If the user tries to access the same panel again, or tries to reload the page, he's shown the 403 error again because he's logged, and the only way for the user to logout is to clean his browser cookies, or go to an authorized panel.

This seems obvious, but the user has reached the admin panel somehow, and I think this can be handled better by simply checking if the user has access to the panel during the credentials validation, something like (`src/Illuminate/Auth/SessionGuard.php:384`):

```php
// If an implementation of UserInterface was returned, we'll ask the provider
// to validate the user against the given credentials, and if they are in
// fact valid and the user can access the current panel, we'll log the
// user into the application and return true.
if (
    $this->hasValidCredentials($user, $credentials) &&
    $user->canAccessPanel(Filament::getCurrentPanel())
) {
    $this->login($user, $remember);

    return true;
}
```

Would you accept a PR to change this behaviour?

**Comment by danharrin** [comment]
Sure 👍
